eoc_journal/base_api_client.py

```diff
@@ -0,0 +1,38 @@
+"""
+A base client for API integrations.
+"""
+from django.conf import settings
+
+from .utils import build_jwt_edx_client
+
+
+# pylint: disable=R0903
+class BaseApiClient(object):
+    """
+    Base API client abstract class.
+    """
+
+    def __init__(self, user, course_id, url=''):
+        """
+        Connect to the REST API.
+        """
+        self.user = user
+        self.course_id = course_id
+        self.expires_in = getattr(settings, 'OAUTH_ID_TOKEN_EXPIRATION', 300)
+        # pylint: disable=C0103
+        self.API_BASE_URL = getattr(settings, 'LMS_ROOT_URL', None)
+        if self.API_BASE_URL:
+            self.API_BASE_URL += url
```

Comment:
I think you should rename this since it's no longer a constant.

Reply:
Also you could just do:
```python
self.api_url = getattr(settings, 'LMS_ROOT_URL', None)
if self.api_url and  and self.API_PATH:
    self.api_url += self.API_PATH
```

Then you don't need to pass the url at all and can get rid of the init method.

Reply:
Neat one!
Refactored.